apps/users/models.py

```diff
@@ -352,7 +352,7 @@ def needs_tougher_password(user):
 
     @property
     def source(self):
-        if self.fxa_id:
+        if self.fxa_id or not self.pk:
```

Comment:
Yeah, I see what you're saying. I'll move it back to the view. I might pass it a stub instead of messing with the user object.